### PR TITLE
CHEF-6493: Support `--legacy-export` option in `inspec archive`

### PR DESCRIPTION
**Pull request by ahasunos**
## Description

This PR adds support for `--legacy-export` flag for including the **inspec.json** file as part of the archive using the information from the legacy export procedure i.e. the results of running `inspec export --legacy-export`.

## Related Issue

CHEF-6493: Modify inspec archive to use `--legacy-export` option

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/profile.rb [hunk 1]
```diff
@@ -808,9 +808,11 @@ def archive(opts)
       # TODO ignore all .files, but add the files to debug output
 
       # Generate temporary inspec.json for archive
-      if opts[:export]
+      export_opt_enabled = opts[:export] || opts[:legacy_export]
+      if export_opt_enabled
+        info_for_profile_summary = opts[:export] ? info_from_parse : info
```
Comment (clintoncwolfe) [inline lib/inspec/profile.rb]: This is wrong - the conditional should be:
```
opts[:legacy_export] ? info : info_from_parse
```

Comment (ahasunos) [reply lib/inspec/profile.rb]: I believe it is not wrong.

https://github.com/inspec/inspec/blob/532e573671ccc1c57a83f2201932cdee2705f9fb/lib/inspec/profile.rb#L813


The above line is checked only if `export_opt_enabled` is true i.e. either the flag `--export` or `--legacy-export` is passed in the cli and is part of the `opts`.

This means the conditions - 
```
opts[:export] ? info_from_parse : info
```
and 
```
opts[:legacy_export] ? info : info_from_parse
```
are both going to equivalent.

Comment (ahasunos) [reply lib/inspec/profile.rb]: Let me know if you mean something else here, which I maybe missing

Comment (ahasunos) [reply lib/inspec/profile.rb]: Addressed!

**Pull request merged**